src/core/reducers/addons.js

```diff
@@ -37,69 +58,151 @@ export function fetchAddon({ errorHandler, slug }: FetchAddonParams): FetchAddon
   };
 }
 
-export function getGuid(result: AddonType) {
-  if (result.type === ADDON_TYPE_THEME) {
+export function getGuid(addon: ExternalAddonType): string {
+  if (addon.type === ADDON_TYPE_THEME) {
     // This mimics how Firefox appends @personas.mozilla.org internally.
     // It's needed to look up themes in mozAddonManager.
-    return `${result.id}@personas.mozilla.org`;
+    return `${addon.id}@personas.mozilla.org`;
   }
-  return result.guid;
+  return addon.guid;
+}
+
+export function removeUndefinedProps(object: Object): Object {
+  const newObject = {};
+  Object.keys(object).forEach((key) => {
+    if (typeof object[key] !== 'undefined') {
+      newObject[key] = object[key];
+    }
+  });
+  return newObject;
 }
 
-export function denormalizeAddon(apiAddon: AddonType) {
-  if (apiAddon.icon_url) {
-    return {
-      ...apiAddon,
-      // Set iconUrl to be consistent between disco and amo.
-      iconUrl: apiAddon.icon_url,
+export function createInternalAddon(
+  apiAddon: ExternalAddonType
+): AddonType {
+  let addon: AddonType = {
+    authors: apiAddon.authors,
+    average_daily_users: apiAddon.average_daily_users,
+    categories: apiAddon.categories,
+    current_beta_version: apiAddon.current_beta_version,
+    current_version: apiAddon.current_version,
+    default_locale: apiAddon.default_locale,
+    description: apiAddon.description,
+    edit_url: apiAddon.edit_url,
+    guid: getGuid(apiAddon),
+    has_eula: apiAddon.has_eula,
+    has_privacy_policy: apiAddon.has_privacy_policy,
+    homepage: apiAddon.homepage,
+    icon_url: apiAddon.icon_url,
+    id: apiAddon.id,
+    is_disabled: apiAddon.is_disabled,
+    is_experimental: apiAddon.is_experimental,
+    is_featured: apiAddon.is_featured,
+    is_source_public: apiAddon.is_source_public,
+    last_updated: apiAddon.last_updated,
+    latest_unlisted_version: apiAddon.latest_unlisted_version,
+    name: apiAddon.name,
+    previews: apiAddon.previews,
+    public_stats: apiAddon.public_stats,
+    ratings: apiAddon.ratings || undefined,
```

Comment:
Why does this one need this?

Reply:
ah, whoops, good catch. That was leftover from me battling with Flow :/